PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * ONNXRuntime.h
+ *
+ *  Created on: Jun 28, 2019
+ *      Author: hqu
+ */
+
+#ifndef PHYSICSTOOLS_ONNXRUNTIME_INTERFACE_ONNXRUNTIME_H_
+#define PHYSICSTOOLS_ONNXRUNTIME_INTERFACE_ONNXRUNTIME_H_
+
+#include <vector>
+#include <map>
+#include <string>
+#include <memory>
+
+#include "onnxruntime/core/session/onnxruntime_cxx_api.h"
+
+namespace Ort {
+
+  typedef std::vector<std::vector<float>> FloatArrays;
+
+  class ONNXRuntime {
+  public:
+    ONNXRuntime();
+    ONNXRuntime(const ONNXRuntime&) = delete;
+    ONNXRuntime& operator=(const ONNXRuntime&) = delete;
+    virtual ~ONNXRuntime();
+
+    // Creates session
+    void createSession(const std::string& model_path, const SessionOptions* session_options = nullptr);
+
+    // Run inference and get outputs
+    // input_names: list of the names of the input nodes.
+    // input_values: list of input arrays for each input node. The order of `input_values` must match `input_names`.
+    // output_names: names of the output nodes to get outputs from. Empty list means all output nodes.
+    // batch_size: number of samples in the batch. Each array in `input_values` must have a shape layout of (batch_size, ...).
+    // Returns: a std::vector<std::vector<float>>, with the order matched to `output_names`.
+    // When `output_names` is empty, will return all outputs ordered as in `getOutputNames()`.
+    FloatArrays run(const std::vector<std::string>& input_names,
+                    FloatArrays input_values,
```

Comment:
Is this vector of vectors intentionally passed by value (leading to copying)?

Reply:
Yes, but thinking again it may not be the best way -- The problem is that `Value::CreateTensor` needs `T *` rather than `const T *`, so here we have to pass a non-const `FloatArrays &` which does not seem very nice to me. Still, I change it to passing by reference, as this object is typically fairly large.